packages/neuron-wallet/src/services/transaction-sender.ts

```diff
@@ -0,0 +1,460 @@
+import { TransactionWithoutHash, WitnessArgs, OutPoint, Cell, Input, DepType } from 'types/cell-types'
+import WalletService, { Wallet } from 'services/wallets'
+import WalletsService from 'services/wallets'
+import { IsRequired } from 'exceptions'
+import NodeService from './node'
+import ConvertTo from 'types/convert-to'
+import { serializeWitnessArgs } from '@nervosnetwork/ckb-sdk-utils'
+import { TransactionPersistor, TransactionsService, TransactionGenerator } from './tx'
+import NetworksService from './networks'
+import { AddressPrefix } from 'models/keys/address'
+import LockUtils from 'models/lock-utils'
+import AddressService from './addresses'
+import { Address } from 'database/address/address-dao'
+import { PathAndPrivateKey } from 'models/keys/key'
+import AddressesService from 'services/addresses'
+import { CellIsNotYetLive, TransactionIsNotCommittedYet } from 'exceptions/dao'
+import TypeConvert from 'types/type-convert'
+import FeeMode from 'models/fee-mode'
+import DaoUtils from 'models/dao-utils'
+import TransactionSize from 'models/transaction-size'
+import TransactionFee from 'models/transaction-fee'
+import logger from 'utils/logger'
+import Keychain from 'models/keys/keychain'
+
+export default class TransactionSender {
+  private static instance: TransactionSender
+
+  private walletService: WalletService
+
+  public static getInstance = () => {
```

Comment:
I feel singleton is unnecessary for this class.

Reply:
Removed